Wootric/WootricSDK-iOS
# Add Opt out to SDK

This PR adds opt out functionality to iOS SDK.

## Changes
- Update smartphone and tablets UI
- Update tests
- Add opt out functionality

## How to test

To enable opt out you should use the newly added method 
```
[Wootric showOptOut:YES];
```
Testing should be straightforward. Just remember the different scenarios:
- Test both smartphones and tablets
- Test selecting all three options:
  - `No, the surveys are fine`
  - `Don't show me a survey for awhile`
  - `Yes, opt me out of all future surveys`

Check the console to see everything is correct
```
a = Account.find_by_account_token 'NPS-1234qwer'
eu = a.end_users.where(email: 'test@email.com').take
ap eu
ap eu.settings
```
Depending on the option you selected, the values should be different.
**No, the surveys are fine:** Nothing should change.
**Don't show me a survey for awhile:** A new decline should be created for the end user.
**Yes, opt me out of all future surveys:** A new decline should be created and the `EndUserSettings` should be something like this
```
#<EndUserSettings:0x000055baaf3eb510> {
                              :id => 71320245,
                     :end_user_id => 71495759,
                       :email_nps => false,
                      :created_at => Wed, 16 May 2018 16:16:36 PDT -07:00,
                      :updated_at => Wed, 16 May 2018 16:16:36 PDT -07:00,
                         :web_nps => false,
                      :mobile_nps => false,
             :force_mobile_survey => nil,
                :force_web_survey => nil,
    :surveys_disabled_by_end_user => true
}
```

- Also check that the opt out button disappears for the social share view (the third view)

## Trello
https://trello.com/c/qc8lu3Xf/16-2-gdpr-ios-sdk

## Note
- I left various commits for easier reviewing. All these will be squashed once this PR has been approved.
